src/types.ts

```diff
@@ -68,6 +68,9 @@ export interface IHistoryTx {
   tokens: string[];
   height?: number;
   processingStatus?: TxHistoryProcessingStatus;
+  nc_id?: string; // For nano contract transaction
+  nc_method?: string; // For nano contract transaction
+  nc_pubkey?: string; // For nano contract transaction
```

Comment:
@pedroferreira1, I believe we need to add it on transactions from fullnode.

I got the transaction list from the fullnode and these informations didn't come with a regular transaction:
```ts
[
  {
    "tx_id": "0000024c0720b64421b8ae7915c24b77291f0d70b78746035b0c98669bf2bd49",
    "version": 4,
    "weight": 21.937001914291383,
    "timestamp": 1709651014,
    "is_voided": false,
    "inputs": [
      {
        "value": 89100,
        "token_data": 1,
        "script": "dqkUUsHGkfofqIoaBi7hbSTJ1pq9TCqIrA==",
        "decoded": {
          "type": "P2PKH",
          "address": "WWDcRRin3Wej7Es1KGmqqVV6RFyyH3x2DQ",
          "timelock": null
        },
        "token": "0000046a7350a07e3985dfbb797fb374706cc7daaf8994765c2ef488730a88fd",
        "tx_id": "000002e1a1a438ea18d2b5bcbddffd549603db8aaddf7ee9f93372de12ea2fce",
        "index": 0
      }
    ],
    "outputs": [
      {
        "value": 88800,
        "token_data": 1,
        "script": "dqkUztu0u/2LY202REnRkKZ0NwdDKzKIrA==",
        "decoded": {
          "type": "P2PKH",
          "address": "WhXoGtGmissCdtTx7ZuCXphZR8ak4JwyVp",
          "timelock": null
        },
        "token": "0000046a7350a07e3985dfbb797fb374706cc7daaf8994765c2ef488730a88fd",
        "spent_by": null
      },
      {
        "value": 300,
        "token_data": 2,
        "script": "dqkUztu0u/2LY202REnRkKZ0NwdDKzKIrA==",
        "decoded": {
          "type": "P2PKH",
          "address": "WhXoGtGmissCdtTx7ZuCXphZR8ak4JwyVp",
          "timelock": null
        },
        "token": "0000000d8f20c7629acdba93f02172bbc7a96e68c27bdc78e03dfe79984d7654",
        "spent_by": null
      }
    ],
    "parents": [
      "000002e1a1a438ea18d2b5bcbddffd549603db8aaddf7ee9f93372de12ea2fce",
      "000000bb7f87575f4a81e71427727c6b56c9575b83d2a8bd2fd79858317f718f"
    ]
  }
]
```

A transaction list from nano contract:
```ts
[
  {
    "hash": "000003c4dcce72f1213186ba7ed54f1f7cc34571d7675f58aeb4a62bc47a29ab",
    "nonce": 1614029,
    "timestamp": 1709650978,
    "version": 4,
    "weight": 21.94525585545934,
    "signal_bits": 0,
    "parents": [
      "0000002ec650e4d058b97fe9662c464116e254f3f41c7244cb7221f61a0c66b2",
      "0000000d8f20c7629acdba93f02172bbc7a96e68c27bdc78e03dfe79984d7654"
    ],
    "inputs": [
      {
        "tx_id": "0000002ec650e4d058b97fe9662c464116e254f3f41c7244cb7221f61a0c66b2",
        "index": 0,
        "data": "RjBEAiAeue30eY/RO/19I+Te4wbE60IDLJm0pZuDlLE4l/0m0AIgYoqlchOB3CfVjpB2IVTh2Jvt3G/fxlfH2TCuK3dQ+NIhA+ET6oRinuICD0kfXAk91AaQFDf+j1ZVxSgDVnNiXZ30"
      }
    ],
    "outputs": [
      {
        "value": 89700,
        "token_data": 1,
        "script": "dqkUnI791s5LV7uuKX/5WyfHsXNt6K+IrA=="
      },
      {
        "value": 300,
        "token_data": 2,
        "script": "dqkUnI791s5LV7uuKX/5WyfHsXNt6K+IrA=="
      }
    ],
    "tokens": [
      "0000046a7350a07e3985dfbb797fb374706cc7daaf8994765c2ef488730a88fd",
      "0000000d8f20c7629acdba93f02172bbc7a96e68c27bdc78e03dfe79984d7654"
    ],
    "nc_id": "0000002ec650e4d058b97fe9662c464116e254f3f41c7244cb7221f61a0c66b2",
    "nc_method": "swap",
    "nc_args": "",
    "nc_pubkey": "024223a7ece4f6f8c64d82db0b30a96411829a3541c6ffee569049af614093192d"
  }
]
```

Reply:
I'm using this information [here](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/434/files#r1514558931).

Reply:
Can you try using the branch from this [PR](https://github.com/HathorNetwork/nano-hathor-core/pull/58), please?

Reply:
Ok. But we still need these info on `TxHistory`.
```
    "nc_id": "0000002ec650e4d058b97fe9662c464116e254f3f41c7244cb7221f61a0c66b2",
    "nc_method": "swap",
    "nc_args": "",
    "nc_pubkey": 
    "024223a7ece4f6f8c64d82db0b30a96411829a3541c6ffee569049af614093192d"
```